### ISSUE TITLE
`tryShutdown` results in cancelled Unary GRPCs

### ISSUE THREAD

**Issue by artificial-aidan** [grpc/grpc-node#2091]
### Problem description

When using `tryShutdown` to gracefully shutdown a server, it doesn't let unary RPCs finish successfully.

### Reproduction steps

1. Create GRPC server with unary handler that takes some extended amount of time
2. Set up a `process.on` handler for `SIGTERM`
3. Call Unary GRPC
4. Before it completes send `SIGTERM` to the server process.
5. Note that client never receives the correct response, just a cancelled call. Even though the server sends the correct data.

### Environment
 - OS name, version and architecture: OSX 10.15.7
 - Node version 17.8.0
 - Package name and version grpc-js@1.3.7

### Additional context

The normal process for a Unary RPC (that I can deduce from debugging) is as follows:

1. `Client` opens stream to `Server`
6. `Client` sends request to `Server`
7. `Server` sends response to `Client`
8. `Client` closes stream.

When using the `tryShutdown` method the process is different.

1. `Client` opens stream to `Server`
3. `Client` sends request to `Server`
4. `tryShutdown` is called
5. `Server` calls `close` on http2 session
9. `Server` sends response to `Client`
10. `Client` see's that stream has been cancelled, and errors.

The cancellation appears to be coming from the session destruction happening in `tryShutdown`. This makes sense for streaming GRPC calls, as they need to be cancelled, but a Unary GRPC should be able to finish its response without the call being cancelled.

Removing the `session.close()` from `tryShutdown` fixes the problem for Unary GRPCs, but obviously doesn't for Streaming GRPCs.

**Comment by murgatroid99** [comment]
It's definitely not supposed to do that, for unary or streaming RPCs. And the [`session.close()` documentation](https://nodejs.org/api/http2.html#http2sessionclosecallback) explicitly says that it doesn't do that, so this sounds like a bug in Node.

Have you tried listening for `SIGUSR2` instead of `SIGTERM`, just to ensure that it's not impacted by some other signal handling? Or even just using a timer to consistently call `tryShutdown` inline? If that's not the issue, would you mind putting together and sharing a complete reproduction of this failure?

**Comment by artificial-aidan** [comment]
I will try `SIGUSR2`. If that doesn't work I will put together something reproducible.

**Comment by artificial-aidan** [comment]
Ok, confirmed with a timer.

https://github.com/artificial-aidan/grpc/tree/aidan/call-cancel

In one terminal run `examples/node/static_codegen/greeter_server.js` and within a few seconds run `greeter_client.js` in another terminal. You will get a call cancelled error on the client.

Edit `server.js` in `node_modules` and comment out the session closing.

Re-run the test. And you will get a successful message.

```
(grpc) Aidans-MBP:node aidan$ node static_codegen/greeter_client.js 
Error: 1 CANCELLED: Call cancelled
    at Object.callErrorFromStatus (/Users/aidan/Development/grpc/examples/node/node_modules/@grpc/grpc-js/build/src/call.js:31:26)
    at Object.onReceiveStatus (/Users/aidan/Development/grpc/examples/node/node_modules/@grpc/grpc-js/build/src/client.js:180:52)
    at Object.onReceiveStatus (/Users/aidan/Development/grpc/examples/node/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:365:141)
    at Object.onReceiveStatus (/Users/aidan/Development/grpc/examples/node/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:328:181)
    at /Users/aidan/Development/grpc/examples/node/node_modules/@grpc/grpc-js/build/src/call-stream.js:187:78
    at processTicksAndRejections (node:internal/process/task_queues:78:11) {
  code: 1,
  details: 'Call cancelled',
  metadata: Metadata { internalRepr: Map(0) {}, options: {} }
}
```

**Comment by murgatroid99** [comment]
On my machine, your example works with earlier versions of Node. It looks like they broke something in Node 16.7.

**Comment by artificial-aidan** [comment]
I was able to confirm the same.

**Comment by murgatroid99** [comment]
I have filed nodejs/node#42713

**Comment by ercang** [comment]
I am having this problem as well, and I couldn't update to the LTS version of Node. I wanted to ask if you have a simpler workaround for this problem as there are not many updates on the Node ticket.

I can think of this;
- divert the traffic from the server
- wait for the maximum timeout
- close the server and hope that all requests ended.

**Comment by aikoven** [comment]
The linked issue was closed but graceful shutdown still doesn't appear to work even on newer NodeJS versions.

**Comment by aikoven** [comment]
Sorry, my minimal reproduction did not show any problem. I'm gonna research further.

**Comment by artificial-aidan** [comment]
Had to upgrade some infra to node 18+, as I read @eplightning's comment, 18.14 should be a safe bet to not have this issue right? I traced through the release notes a bit and think I found that as well, but just double checking.

**Comment by murgatroid99** [comment]
This has been fixed in Node 20.10.0.
